src/domain/runner/instance_wrapper.rs

```diff
@@ -94,19 +99,19 @@ impl InstanceWrapper {
         self.set_remaining_gas(store, gas_after);
     }
 
-    pub fn refund_gas(&self, store: &mut impl AsStoreMut, refund: u64) {
-        let gas_before = self.get_remaining_gas(store);
+    pub fn refund_gas(&self, store: &mut impl AsStoreMut, refund: i64) {
```

Comment:
Ethereum refunds gas at the end of a transaction.
This method is principally the same as use_gas.

Reply:
If the refund is "negative", I can call another `use_gas` instead - it will be cleaner